pkg/steps/bundle_source.go

```diff
@@ -0,0 +1,125 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	buildapi "github.com/openshift/api/build/v1"
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/results"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	coreapi "k8s.io/api/core/v1"
+)
+
+type bundleSourceStep struct {
+	config      api.BundleSourceStepConfiguration
+	resources   api.ResourceConfiguration
+	buildClient BuildClient
+	imageClient imageclientset.ImageStreamsGetter
+	istClient   imageclientset.ImageStreamTagsGetter
+	jobSpec     *api.JobSpec
+	artifactDir string
+	dryLogger   *DryLogger
+	pullSecret  *coreapi.Secret
+}
+
+func (s *bundleSourceStep) Inputs(dry bool) (api.InputDefinition, error) {
+	return nil, nil
+}
+
+func (s *bundleSourceStep) Run(ctx context.Context, dry bool) error {
+	return results.ForReason("building_bundle_source").ForError(s.run(ctx, dry))
+}
+
+func (s *bundleSourceStep) run(ctx context.Context, dry bool) error {
+	source := fmt.Sprintf("%s:%s", api.PipelineImageStream, api.PipelineImageStreamTagReferenceSource)
+	var workingDir string
+	if dry {
+		workingDir = "dry-fake"
+	} else {
+		var err error
+		workingDir, err = getWorkingDir(s.istClient, source, s.jobSpec.Namespace())
+		if err != nil {
+			return fmt.Errorf("failed to get workingDir: %v", err)
+		}
+	}
+	dockerfile := bundleSourceDockerfile(s.config.OperatorManifests, s.config.Substitute)
+	build := buildFromSource(
+		s.jobSpec, api.PipelineImageStreamTagReferenceSource, s.config.To,
+		buildapi.BuildSource{
+			Type:       buildapi.BuildSourceDockerfile,
+			Dockerfile: &dockerfile,
+			Images: []buildapi.ImageSource{
+				{
+					From: coreapi.ObjectReference{
+						Kind: "ImageStreamTag",
+						Name: source,
+					},
+					Paths: []buildapi.ImageSourcePath{{
+						SourcePath:     fmt.Sprintf("%s/%s/.", workingDir, s.config.ContextDir),
+						DestinationDir: ".",
+					}},
+				},
+			},
+		},
+		"",
+		s.resources,
+		s.pullSecret,
+	)
+	return handleBuild(ctx, s.buildClient, build, dry, s.artifactDir, s.dryLogger)
+}
+
+func replaceCommand(manifestDir, pullSpec, with string) string {
+	return fmt.Sprintf("find %s -type f -exec sed -i 's?%s?stable:%s?g' {} +", manifestDir, pullSpec, with)
```

Comment:
Hmm... This replaces the pullspec with `stable:with`, which doesn't seem to be enough, we need to have a pullspec (whatever we set `IMAGE_$with` to).

Reply:
Changed it so it uses the full registry pullspec